scene/animation/animation_blend_tree.cpp

```diff
@@ -734,26 +742,27 @@ double AnimationNodeTransition::process(double p_time, bool p_seek, bool p_seek_
 			time += p_time;
 		}
 
-		if (inputs[current].auto_advance && rem <= xfade) {
+		if (inputs[current].auto_advance && rem <= xfade_time) {
 			set_parameter(this->current, (current + 1) % enabled_inputs);
 		}
 
 	} else { // cross-fading from prev to current
 
-		float blend = xfade == 0 ? 0 : (prev_xfading / xfade);
+		float blend = xfade_time == 0 ? 0 : (prev_xfading / xfade_time);
+		float curved_blend = (xfade_curve.is_valid()) ? xfade_curve->interpolate(1.0 - blend) : 1.0 - blend;
 
 		if (from_start && !p_seek && switched) { //just switched, seek to start of current
-
-			rem = blend_input(current, 0, true, p_seek_root, 1.0 - blend, FILTER_IGNORE, true);
+			rem = blend_input(current, 0, true, p_seek_root, curved_blend, FILTER_IGNORE, true);
 		} else {
-			rem = blend_input(current, p_time, p_seek, p_seek_root, 1.0 - blend, FILTER_IGNORE, true);
+			rem = blend_input(current, p_time, p_seek, p_seek_root, curved_blend, FILTER_IGNORE, true);
 		}
 
+		curved_blend = (xfade_curve.is_valid()) ? xfade_curve->interpolate(blend) : blend;
 		if (p_seek) {
-			blend_input(prev, p_time, true, p_seek_root, blend, FILTER_IGNORE, true);
+			blend_input(prev, p_time, true, p_seek_root, curved_blend, FILTER_IGNORE, true);
 			time = p_time;
 		} else {
-			blend_input(prev, p_time, false, p_seek_root, blend, FILTER_IGNORE, true);
+			blend_input(prev, p_time, false, p_seek_root, curved_blend, FILTER_IGNORE, true);
```

Comment:
Suggestion:
```cpp
	} else { // cross-fading from prev to current

		float blend = xfade_time == 0 ? 0 : (prev_xfading / xfade_time);
		if (xfade_curve.is_valid()) {
			blend = xfade_curve->interpolate(blend);
		}

		if (from_start && !p_seek && switched) { //just switched, seek to start of current

			rem = blend_input(current, 0, true, p_seek_root, 1.0 - blend, FILTER_IGNORE, true);
		} else {
			rem = blend_input(current, p_time, p_seek, p_seek_root, 1.0 - blend, FILTER_IGNORE, true);
		}

		if (p_seek) {
			blend_input(prev, p_time, true, p_seek_root, blend, FILTER_IGNORE, true);
			time = p_time;
		} else {
			blend_input(prev, p_time, false, p_seek_root, blend, FILTER_IGNORE, true);
			time += p_time;
			prev_xfading -= p_time;
			if (prev_xfading < 0) {
				set_parameter(this->prev, -1);
			}
		}
	}
```
Your interpolation calculation is wrong; you only need to overwrite blend with curve.